### PR TITLE
Feature - 2790 - Add New Search Container/Form

### LINKED ISSUE #2790: SearchContainerApi 2.0 component
In SearchPage component, if APPLICANTSEARCH feature flag is true, render a new component instead of SearchContainerApi. This new component will be essentially identical to existing SearchContainerApi, but will be connected to a different API query, and use a different SearchForm component (which itself will start as almost identical to the current SearchForm component). 

The SearchContainer component may be refactor to accept SearchForm as children, allowing it to be used as a wrapper around both versions.

Rename the current SearchContainerApi and SearchForm to SearchContainerApiOld and SearchFormOld, because they will eventually be deleted in favour of the new versions.

## Acceptance Criteria
- [x] Renders different components based on feature flag
- [x] Old components renamed, project refactor to match
- [x] Create new APPLICANTSEARCH feature flag.

### PR DESCRIPTION
**Pull request by esizer**
Resolves #2790 

## Summary

This duplicates the existing `SearchContainer` and `SearchForm` putting them behind the feature flag `FEATURE_APPLICANTSEARCH`.

## Testing

1. Rebuild your php container
2. Build the `talentsearch` frontend
3. Navigate to `/talent/search`
4. Test the form (it should function no differently)

### CODE REVIEW

**Comment by petertgiles** [comment]
LOL, there's a bit more than just copying the component going on here. :laughing: Could you give me a guide to the refactoring that happened here to help me follow?

**Comment by esizer** [comment]
> LOL, there's a bit more than just copying the component going on here. 😆 Could you give me a guide to the refactoring that happened here to help me follow?

Yes, git did some interesting things with +/- that I don't understand which makes it seem more complicated than it really is.

- Extracted [CandidateResults](https://github.com/GCTC-NTGC/gc-digital-talent/blob/8c75d14458e8633b1c06eaa2cb623bdd78090cf0/frontend/talentsearch/src/js/components/search/CandidateResults.tsx) into own file.
- Extracted [FilterBlock](https://github.com/GCTC-NTGC/gc-digital-talent/blob/8c75d14458e8633b1c06eaa2cb623bdd78090cf0/frontend/talentsearch/src/js/components/search/FilterBlock.tsx) into own file
- Updated function props to use a more conventional `on` prefix
- Removed `updateInitialValues` since it was not in use
- Updated to prefer default exports

**Review by petertgiles — APPROVED** [review]
Very nice!

File: frontend/talentsearch/src/js/components/search/SearchContainer.tsx [hunk 1]
```diff
@@ -203,62 +181,32 @@ export const SearchContainer: React.FC<SearchContainerProps> = ({
...
-        pools: poolId ? [{ id: poolId }] : pickMap(filter?.pools, "id"),
-      },
-    };
-  return {};
-};
-
-export const SearchContainerApi: React.FC = () => {
-  const [initialValues, setInitialValues] = useState<FormValues | null>(null);
+const SearchContainerApi: React.FC = () => {
+  const [initialValues, setInitialValues] = React.useState<FormValues | null>(
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/talentsearch/src/js/components/search/SearchContainer.tsx]: removed with[remove unused search state](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3109/commits/560619be4195961b055706006e3c8ffc79d13e58)

**Pull request merged**